### ISSUE TITLE
[bug] Conan doesn't generate required build information

### ISSUE THREAD

**Issue by HappyCerberus** [conan-io/conan#13234]
### Environment details

* Operating System+version: WSL2 / Ubuntu 22.04.2 LTS
* Compiler+version: Ubuntu clang version 13.0.1-2ubuntu2.1
* Conan version: Conan version 1.59.0
* Python version: Python 3.10.6

### Steps to reproduce

https://github.com/HappyCerberus/conan-experiment

```
pushd conan
conan install ../conanfile.txt
popd
bazel build :hello
```

Fails with `Repository '@opengl' is not defined and referenced by '@sdl//:sdl'`

Which is indeed correct:

```
$ cat conan/sdl/BUILD
load("@rules_cc//cc:defs.bzl", "cc_import", "cc_library")

cc_import(
    name = "SDL2main_precompiled",
    static_library = "lib/libSDL2main.a",
)

cc_import(
    name = "SDL2_precompiled",
    static_library = "lib/libSDL2.a",
)

cc_library(
    name = "sdl",
    hdrs = glob(["include/**", "include/SDL2/**"]),
    includes = ["include", "include/SDL2"],
    linkopts = ["-ldl", "-lrt", "-lpthread"],   
    visibility = ["//visibility:public"],
    deps = [
        # do not sort
    ":SDL2main_precompiled",
    ":SDL2_precompiled",
    "@libiconv",
    "@libalsa",
    "@pulseaudio",
    "@opengl",
    "@nas",
    "@wayland",
    "@xkbcommon",
    "@egl",
    "@libunwind",
    ],   
)
```

and:

```
$ cat conan/dependencies.bzl
def load_conan_dependencies():
    
    native.new_local_repository(
        name="sdl",
        path="/home/simon/.conan/data/sdl/2.26.1/_/_/package/e31ead8f4f2fae622ce4af90ebc3f92dfbab9eb9",
        build_file="/home/simon/conan-experiment/conan/sdl/BUILD",
    )
    
    native.new_local_repository(
        name="libiconv",
        path="/home/simon/.conan/data/libiconv/1.17/_/_/package/63c2a85d57849e261f98f935b93ecac31ba71b84",
        build_file="/home/simon/conan-experiment/conan/libiconv/BUILD",
    )
    
    native.new_local_repository(
        name="libalsa",
        path="/home/simon/.conan/data/libalsa/1.2.7.2/_/_/package/415acf4d80e82107880a90fe05ab6812dd715145",
        build_file="/home/simon/conan-experiment/conan/libalsa/BUILD",
    )
    
    native.new_local_repository(
        name="pulseaudio",
        path="/home/simon/.conan/data/pulseaudio/14.2/_/_/package/1cf14d58fd3730a3ffe6205a36f5d49b0251bd95",
        build_file="/home/simon/conan-experiment/conan/pulseaudio/BUILD",
    )
    
    native.new_local_repository(
        name="nas",
        path="/home/simon/.conan/data/nas/1.9.4/_/_/package/2ec500fddc396c58a48ca6f95090d46a7f97968c",
        build_file="/home/simon/conan-experiment/conan/nas/BUILD",
    )
    
    native.new_local_repository(
        name="wayland",
        path="/home/simon/.conan/data/wayland/1.21.0/_/_/package/2e5d593200c8e1a7c42eb3bb220fce94da2a0f3a",
        build_file="/home/simon/conan-experiment/conan/wayland/BUILD",
    )
    
    native.new_local_repository(
        name="xkbcommon",
        path="/home/simon/.conan/data/xkbcommon/1.4.1/_/_/package/564afa6b7157c454afc5b18164bf7daea2eaa62e",
        build_file="/home/simon/conan-experiment/conan/xkbcommon/BUILD",
    )
    
    native.new_local_repository(
        name="libunwind",
        path="/home/simon/.conan/data/libunwind/1.6.2/_/_/package/daae54c63b7c37ccc1d2ccf388ecb125051c9131",
        build_file="/home/simon/conan-experiment/conan/libunwind/BUILD",
    )
    
    native.new_local_repository(
        name="libsndfile",
        path="/home/simon/.conan/data/libsndfile/1.0.31/_/_/package/5536afc05c3ed33dc8a48c82b37c247935616e48",
        build_file="/home/simon/conan-experiment/conan/libsndfile/BUILD",
    )
    
    native.new_local_repository(
        name="libcap",
        path="/home/simon/.conan/data/libcap/2.65/_/_/package/63c2a85d57849e261f98f935b93ecac31ba71b84",
        build_file="/home/simon/conan-experiment/conan/libcap/BUILD",
    )
    
    native.new_local_repository(
        name="libtool",
        path="/home/simon/.conan/data/libtool/2.4.7/_/_/package/2782bfb6d1ecbdcb96073111f080fabaf976642f",
        build_file="/home/simon/conan-experiment/conan/libtool/BUILD",
    )
    
    native.new_local_repository(
        name="xorg",
        path="/home/simon/.conan/data/xorg/system/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9",
        build_file="/home/simon/conan-experiment/conan/xorg/BUILD",
    )
    
    native.new_local_repository(
        name="openssl",
        path="/home/simon/.conan/data/openssl/1.1.1q/_/_/package/63c2a85d57849e261f98f935b93ecac31ba71b84",
        build_file="/home/simon/conan-experiment/conan/openssl/BUILD",
    )
    
    native.new_local_repository(
        name="libffi",
        path="/home/simon/.conan/data/libffi/3.4.3/_/_/package/63c2a85d57849e261f98f935b93ecac31ba71b84",
        build_file="/home/simon/conan-experiment/conan/libffi/BUILD",
    )
    
    native.new_local_repository(
        name="libxml2",
        path="/home/simon/.conan/data/libxml2/2.10.3/_/_/package/9fdc908aebad182c6517a0090e1f564bff8bfad7",
        build_file="/home/simon/conan-experiment/conan/libxml2/BUILD",
    )
    
    native.new_local_repository(
        name="expat",
        path="/home/simon/.conan/data/expat/2.5.0/_/_/package/bef71e5238b7b03b858f1fa7a6f20fae013f892b",
        build_file="/home/simon/conan-experiment/conan/expat/BUILD",
    )
    
    native.new_local_repository(
        name="xz_utils",
        path="/home/simon/.conan/data/xz_utils/5.4.0/_/_/package/63c2a85d57849e261f98f935b93ecac31ba71b84",
        build_file="/home/simon/conan-experiment/conan/xz_utils/BUILD",
    )
    
    native.new_local_repository(
        name="zlib",
        path="/home/simon/.conan/data/zlib/1.2.13/_/_/package/63c2a85d57849e261f98f935b93ecac31ba71b84",
        build_file="/home/simon/conan-experiment/conan/zlib/BUILD",
    )
    
    native.new_local_repository(
        name="ogg",
        path="/home/simon/.conan/data/ogg/1.3.5/_/_/package/63c2a85d57849e261f98f935b93ecac31ba71b84",
        build_file="/home/simon/conan-experiment/conan/ogg/BUILD",
    )
    
    native.new_local_repository(
        name="vorbis",
        path="/home/simon/.conan/data/vorbis/1.3.7/_/_/package/9103036656e484d65af0b185ee201208be47715d",
        build_file="/home/simon/conan-experiment/conan/vorbis/BUILD",
    )
    
    native.new_local_repository(
        name="flac",
        path="/home/simon/.conan/data/flac/1.4.2/_/_/package/0f7673ed046dfe07dc3817a05cf1aaf246ef53e1",
        build_file="/home/simon/conan-experiment/conan/flac/BUILD",
    )
    
    native.new_local_repository(
        name="opus",
        path="/home/simon/.conan/data/opus/1.3.1/_/_/package/63c2a85d57849e261f98f935b93ecac31ba71b84",
        build_file="/home/simon/conan-experiment/conan/opus/BUILD",
    )
```

### Logs

```
conan install ../conanfile.txt 
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=clang
compiler.libcxx=libc++
compiler.version=13
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

conanfile.txt: Installing package
Requirements
    autoconf/2.71 from 'conancenter' - Cache
    automake/1.16.5 from 'conancenter' - Cache
    egl/system from 'conancenter' - Cache
    expat/2.5.0 from 'conancenter' - Cache
    flac/1.4.2 from 'conancenter' - Cache
    libalsa/1.2.7.2 from 'conancenter' - Cache
    libcap/2.65 from 'conancenter' - Cache
    libffi/3.4.3 from 'conancenter' - Cache
    libiconv/1.17 from 'conancenter' - Cache
    libsndfile/1.0.31 from 'conancenter' - Cache
    libtool/2.4.7 from 'conancenter' - Cache
    libunwind/1.6.2 from 'conancenter' - Cache
    libxml2/2.10.3 from 'conancenter' - Cache
    m4/1.4.19 from 'conancenter' - Cache
    nas/1.9.4 from 'conancenter' - Cache
    ogg/1.3.5 from 'conancenter' - Cache
    opengl/system from 'conancenter' - Cache
    openssl/1.1.1q from 'conancenter' - Cache
    opus/1.3.1 from 'conancenter' - Cache
    pulseaudio/14.2 from 'conancenter' - Cache
    sdl/2.26.1 from 'conancenter' - Cache
    vorbis/1.3.7 from 'conancenter' - Cache
    wayland/1.21.0 from 'conancenter' - Cache
    wayland-protocols/1.31 from 'conancenter' - Cache
    xkbcommon/1.4.1 from 'conancenter' - Cache
    xkeyboard-config/system from 'conancenter' - Cache
    xorg/system from 'conancenter' - Cache
    xz_utils/5.4.0 from 'conancenter' - Cache
    zlib/1.2.13 from 'conancenter' - Cache
Packages
    autoconf/2.71:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    automake/1.16.5:258f6f05ca54813b3d7180161753df7f9ccf7e1d - Cache
    egl/system:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    expat/2.5.0:bef71e5238b7b03b858f1fa7a6f20fae013f892b - Cache
    flac/1.4.2:0f7673ed046dfe07dc3817a05cf1aaf246ef53e1 - Cache
    libalsa/1.2.7.2:415acf4d80e82107880a90fe05ab6812dd715145 - Cache
    libcap/2.65:63c2a85d57849e261f98f935b93ecac31ba71b84 - Cache
    libffi/3.4.3:63c2a85d57849e261f98f935b93ecac31ba71b84 - Cache
    libiconv/1.17:63c2a85d57849e261f98f935b93ecac31ba71b84 - Cache
    libsndfile/1.0.31:5536afc05c3ed33dc8a48c82b37c247935616e48 - Cache
    libtool/2.4.7:2782bfb6d1ecbdcb96073111f080fabaf976642f - Cache
    libunwind/1.6.2:daae54c63b7c37ccc1d2ccf388ecb125051c9131 - Cache
    libxml2/2.10.3:9fdc908aebad182c6517a0090e1f564bff8bfad7 - Cache
    m4/1.4.19:24647d9fe8ec489125dfbae4b3ebefaf7581674c - Cache
    nas/1.9.4:2ec500fddc396c58a48ca6f95090d46a7f97968c - Cache
    ogg/1.3.5:63c2a85d57849e261f98f935b93ecac31ba71b84 - Cache
    opengl/system:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    openssl/1.1.1q:63c2a85d57849e261f98f935b93ecac31ba71b84 - Cache
    opus/1.3.1:63c2a85d57849e261f98f935b93ecac31ba71b84 - Cache
    pulseaudio/14.2:1cf14d58fd3730a3ffe6205a36f5d49b0251bd95 - Cache
    sdl/2.26.1:e31ead8f4f2fae622ce4af90ebc3f92dfbab9eb9 - Cache
    vorbis/1.3.7:9103036656e484d65af0b185ee201208be47715d - Cache
    wayland/1.21.0:2e5d593200c8e1a7c42eb3bb220fce94da2a0f3a - Cache
    wayland-protocols/1.31:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    xkbcommon/1.4.1:564afa6b7157c454afc5b18164bf7daea2eaa62e - Cache
    xkeyboard-config/system:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    xorg/system:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    xz_utils/5.4.0:63c2a85d57849e261f98f935b93ecac31ba71b84 - Cache
    zlib/1.2.13:63c2a85d57849e261f98f935b93ecac31ba71b84 - Cache

Installing (downloading, building) binaries...
egl/system: Already installed!
expat/2.5.0: Already installed!
libalsa/1.2.7.2: Already installed!
libcap/2.65: Already installed!
libffi/3.4.3: Already installed!
libiconv/1.17: Already installed!
m4/1.4.19: Already installed!
m4/1.4.19: Appending PATH environment variable: /home/simon/.conan/data/m4/1.4.19/_/_/package/24647d9fe8ec489125dfbae4b3ebefaf7581674c/bin
ogg/1.3.5: Already installed!
opengl/system: Already installed!
openssl/1.1.1q: Already installed!
opus/1.3.1: Already installed!
wayland-protocols/1.31: Already installed!
xkeyboard-config/system: Already installed!
xorg/system: Already installed!
xz_utils/5.4.0: Already installed!
zlib/1.2.13: Already installed!
autoconf/2.71: Already installed!
autoconf/2.71: Defining AC_MACRODIR environment variable: /home/simon/.conan/data/autoconf/2.71/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/res/autoconf
autoconf/2.71: Defining autom4te_perllibdir environment variable: /home/simon/.conan/data/autoconf/2.71/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/res/autoconf
autoconf/2.71: Defining AUTOCONF environment variable: /home/simon/.conan/data/autoconf/2.71/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/bin/autoconf
autoconf/2.71: Defining AUTORECONF environment variable: /home/simon/.conan/data/autoconf/2.71/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/bin/autoreconf
autoconf/2.71: Defining AUTOHEADER environment variable: /home/simon/.conan/data/autoconf/2.71/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/bin/autoheader
autoconf/2.71: Defining AUTOM4TE environment variable: /home/simon/.conan/data/autoconf/2.71/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/bin/autom4te
flac/1.4.2: Already installed!
flac/1.4.2: Appending PATH environment variable: /home/simon/.conan/data/flac/1.4.2/_/_/package/0f7673ed046dfe07dc3817a05cf1aaf246ef53e1/bin
libunwind/1.6.2: Already installed!
libxml2/2.10.3: Already installed!
libxml2/2.10.3: Appending PATH environment variable: /home/simon/.conan/data/libxml2/2.10.3/_/_/package/9fdc908aebad182c6517a0090e1f564bff8bfad7/bin
nas/1.9.4: Already installed!
nas/1.9.4: Appending PATH environment variable: /home/simon/.conan/data/nas/1.9.4/_/_/package/2ec500fddc396c58a48ca6f95090d46a7f97968c/bin
vorbis/1.3.7: Already installed!
automake/1.16.5: Already installed!
libsndfile/1.0.31: Already installed!
libsndfile/1.0.31: Appending PATH environment variable: /home/simon/.conan/data/libsndfile/1.0.31/_/_/package/5536afc05c3ed33dc8a48c82b37c247935616e48/bin
wayland/1.21.0: Already installed!
libtool/2.4.7: Already installed!
libtool/2.4.7: Appending PATH env: /home/simon/.conan/data/libtool/2.4.7/_/_/package/2782bfb6d1ecbdcb96073111f080fabaf976642f/bin
libtool/2.4.7: Setting LIBTOOLIZE env to /home/simon/.conan/data/libtool/2.4.7/_/_/package/2782bfb6d1ecbdcb96073111f080fabaf976642f/bin/libtoolize
libtool/2.4.7: Setting LIBTOOL_PREFIX environment variable to /home/simon/.conan/data/libtool/2.4.7/_/_/package/2782bfb6d1ecbdcb96073111f080fabaf976642f
libtool/2.4.7: Setting LIBTOOL_DATADIR environment variable to /home/simon/.conan/data/libtool/2.4.7/_/_/package/2782bfb6d1ecbdcb96073111f080fabaf976642f/res
libtool/2.4.7: Setting LIBTOOL_PKGAUXDIR environment variable to /home/simon/.conan/data/libtool/2.4.7/_/_/package/2782bfb6d1ecbdcb96073111f080fabaf976642f/res/libtool/build-aux
libtool/2.4.7: Setting LIBTOOL_PKGLTDLDIR environment variable to /home/simon/.conan/data/libtool/2.4.7/_/_/package/2782bfb6d1ecbdcb96073111f080fabaf976642f/res/libtool
libtool/2.4.7: Setting LIBTOOL_ACLOCALDIR environment variable to /home/simon/.conan/data/libtool/2.4.7/_/_/package/2782bfb6d1ecbdcb96073111f080fabaf976642f/res/aclocal
libtool/2.4.7: Appending ACLOCAL_PATH env: /home/simon/.conan/data/libtool/2.4.7/_/_/package/2782bfb6d1ecbdcb96073111f080fabaf976642f/res/aclocal
libtool/2.4.7: Appending AUTOMAKE_CONAN_INCLUDES environment variable: /home/simon/.conan/data/libtool/2.4.7/_/_/package/2782bfb6d1ecbdcb96073111f080fabaf976642f/res/aclocal
xkbcommon/1.4.1: Already installed!
xkbcommon/1.4.1: Appending PATH environment variable: /home/simon/.conan/data/xkbcommon/1.4.1/_/_/package/564afa6b7157c454afc5b18164bf7daea2eaa62e/bin
pulseaudio/14.2: Already installed!
sdl/2.26.1: Already installed!
conanfile.txt: WARN: Using the new toolchains and generators without specifying a build profile (e.g: -pr:b=default) is discouraged and might cause failures and unexpected behavior
conanfile.txt: Generator 'BazelDeps' calling 'generate()'
conanfile.txt: WARN: Using the new toolchains and generators without specifying a build profile (e.g: -pr:b=default) is discouraged and might cause failures and unexpected behavior
conanfile.txt: Generator 'BazelToolchain' calling 'generate()'
conanfile.txt: Generator txt created conanbuildinfo.txt
conanfile.txt: Aggregating env generators
conanfile.txt: Generated conaninfo.txt
conanfile.txt: Generated graphinfo
```

**Comment by memsharded** [comment]
Hi @HappyCerberus 

Thanks for your report.
One of the things that we always struggle is that we have little bazel experience in the team.

I think this could be caused by ``opengl`` being a ConanCenter "system" dependency, which means that it doesn't really compile or contain any binary, but it is just a proxy/wrapper of the system libraries, to help model them. And the Bazel generators not correctly managing that. But there are also other ones, like gle or xorg that are "system" too, do they suffer the same issue?

If you could give some hints about how the generated build files should be or what could be the missing bits there, that could help. Thanks!

**Comment by HappyCerberus** [comment]
@memsharded Not that I'm a Bazel expert, but I think the simplest approach would be to generate a BUILD file for every dependency, including those installed on the system.

There is a deeper philosophical question of how exactly a binary package manager should work with an at HEAD build system like Bazel, considering Bazel takes a lot of effort to make builds hermetic and reproducible. But that is something to discuss with Bazel developers.

**Comment by franramirez688** [comment]
I checked this issue from my branch https://github.com/conan-io/conan/pull/14959 and it seems solved.
I say "seems" because I'm getting new errors but it looks like they could be related to the recipes themselves:

```
$ bazel build :hello
INFO: Analyzed target //:hello (5 packages loaded, 176 targets configured).
INFO: Found 1 target...
ERROR: /Users/franchuti/develop/conan/mmmm/conan-experiment/BUILD:1:10: Linking hello failed: (Exit 1): cc_wrapper.sh failed: error executing command (from target //:hello) external/local_config_cc/cc_wrapper.sh @bazel-out/darwin-fastbuild/bin/hello-2.params

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
ld: warning: ignoring duplicate libraries: '-lc++'
ld: Undefined symbols:
  _AudioObjectAddPropertyListener, referenced from:
      _COREAUDIO_Init in libSDL2.a[139](SDL_coreaudio.m.o)
      _audioqueue_thread in libSDL2.a[139](SDL_coreaudio.m.o)
      _audioqueue_thread in libSDL2.a[139](SDL_coreaudio.m.o)
  _AudioObjectGetPropertyData, referenced from:
      _COREAUDIO_GetDefaultAudioInfo in libSDL2.a[139](SDL_coreaudio.m.o)
      _COREAUDIO_GetDefaultAudioInfo in libSDL2.a[139](SDL_coreaudio.m.o)
      _COREAUDIO_GetDefaultAudioInfo in libSDL2.a[139](SDL_coreaudio.m.o)
      _COREAUDIO_GetDefaultAudioInfo in libSDL2.a[139](SDL_coreaudio.m.o)
      _prepare_device in libSDL2.a[139](SDL_coreaudio.m.o)
      _prepare_device in libSDL2.a[139](SDL_coreaudio.m.o)
      _prepare_device in libSDL2.a[139](SDL_coreaudio.m.o)
      ...
      _Cocoa_AcceptDragAndDrop in libSDL2.a[175](SDL_cocoawindow.m.o)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Error in child process '/usr/bin/xcrun'. 1
Target //:hello failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 23.433s, Critical Path: 1.89s
INFO: 6 processes: 5 internal, 1 darwin-sandbox.
FAILED: Build did NOT complete successfully
```
Overall, the original issue is solved but it might be another kind of problem coming from the recipe declaration.

**Comment by franramirez688** [comment]
Closed by https://github.com/conan-io/conan/pull/14959

It'll be released in the next Conan v1.62.
